tests/modeling/test_roi_heads.py

```diff
@@ -131,6 +133,41 @@ def test_rroi_heads(self):
             ),
         )
 
+    @unittest.skipIf(TORCH_VERSION < (1, 7), "Insufficient pytorch version")
+    def test_mask_head_scriptability(self):
+        cfg = get_cfg()
+        cfg.MODEL.ROI_MASK_HEAD.NUM_CONV = 1
+        cfg.MODEL.MASK_ON = True
+
+        input_shape = ShapeSpec(channels=1024)
+
+        mask_features = torch.randn(4, 1024, 14, 14)
+
+        image_shapes = [(10, 10), (15, 15)]
+        pred_instance0 = Instances(image_shapes[0])
+        pred_classes0 = torch.tensor([1, 2, 3], dtype=torch.int64)
+        pred_instance0.pred_classes = pred_classes0
+        pred_instance1 = Instances(image_shapes[1])
+        pred_classes1 = torch.tensor([4], dtype=torch.int64)
+        pred_instance1.pred_classes = pred_classes1
+
+        mask_head = build_mask_head(cfg, input_shape).eval()
+        origin_outputs = mask_head(mask_features, [pred_instance0, pred_instance1])
+
+        fields = {"pred_masks": "Tensor", "pred_classes": "Tensor"}
+        with patch_instances(fields) as new_instance:
+            sciript_mask_head = torch.jit.script(mask_head)
+            pred_instance0 = new_instance(image_shapes[0])
+            pred_instance0.pred_classes = pred_classes0
```

Comment:
I think adding a conversion staticmethod `NewInstances.from_instances(instances: Instances) -> NewInstances` would be useful. This method is `@jit.unused` and is only meant for users to construct inputs for modules that need `NewInstances` as input.

Reply:
done